pkg/controller/servicemesh/controlplane/deleter.go

```diff
@@ -89,12 +92,15 @@ func (r *controlPlaneInstanceReconciler) Delete(ctx context.Context) error {
 			log.Info("Removed finalizer")
 			hacks.SkipReconciliationUntilCacheSynced(ctx, smcpNamespacedName)
 		} else if !apierrors.IsNotFound(err) {
-			r.EventRecorder.Event(instance, corev1.EventTypeWarning, eventReasonFailedRemovingFinalizer, fmt.Sprintf("Error occurred removing finalizer from service mesh: %s", err)) // TODO: this event probably isn't needed at all
-			return errors.Wrap(err, "Error removing ServiceMeshControlPlane finalizer")
+			// TODO: this event probably isn't needed at all
```

Comment:
worth to create an issue?

Reply:
I'll probably just remove the event the next time I make changes here.